### PR TITLE
List functions for Total, Accumulate, FoldList, Fold

### PR DESCRIPTION
**Pull request by wolfv**
I added some functions to lists that were missing from Mathics. 

Unfortunately, the `Flatten` doesn't work the same way as in Mathematica. Therefore, Total is incomplete. 

Maybe someone with more knowledge of Mathics can fix that easily.

Cheers,

Wolf

### CODE REVIEW

File: mathics/builtin/lists.py [hunk 1]
```diff
@@ -2012,3 +2012,172 @@ def get_call():
...
+    <dt>Total[$list$]'
+        <dd>
+        Add all values up to calculate total
+        Equivalent to Fold[Plus, $list$] or Apply[Plus, $list$]
+        </dd>
+        </dt>
+    </dl>
+    >> Total[{1,2,3}]
+     = 6
+    """
```
Comment (sn6uv) [inline mathics/builtin/lists.py]: Is this where we need the proper `Flatten` functionality; for the other `Total` form?

Comment (wolfv) [inline mathics/builtin/lists.py]: Yep!

Comment (wolfv) [inline mathics/builtin/lists.py]: Then you can sum along rows or along columns etc.

**Comment by sn6uv** [comment]
Looks really good.The only thingis, it looks like there are some duplicate class definitions, e.g. `Total` is defined twice.

**Comment by wolfv** [comment]
Whoops, look like a mistake, sorry. 

Do you think the flatten is easy to do? Let me know if I can help you out somehow. I can give it a try if you give me the right pointers.

Cheers,

Wolf

**Comment by sn6uv** [comment]
I was about to ask for an example but I see there's one in #251. I'll look into it.

**Comment by sn6uv** [comment]
I've just merged #253. @wolfv are you intending to make more changes to this PR?

File: mathics/builtin/lists.py (hunk 1 above)
Comment (wolfv) [inline mathics/builtin/lists.py]: Works now:
```
In[7]:= Total[{1,2,3}]
Out[7]= 6

In[8]:= Total[a,{{2},{1}}]
Out[8]= {6, 15, 24}

In[9]:= Total[a,{{1},{2}}]
Out[9]= {12, 15, 18}
```
:+1:

**Comment by wolfv** [comment]
Argh, something got mixed up in the commits. I added some documentation but I can't really formulate the FoldList better. I hope that it's clear from the examples what it does...

Cool stuff on the flatten! Works perfect in combination with `Total` now!

**Comment by wolfv** [comment]
I just created a new branch in my repository that should have all the initial changes plus new documentation but without the indexing changes (probably better as a seperate merge).

It's here: https://github.com/wolfv/Mathics/tree/rebased-listfold

**Pull request closed without merging**